test/unit/codegen/codegen_cpp_visitor.cpp

```diff
@@ -1083,6 +1083,41 @@ SCENARIO("Some tests on derivimplicit", "[codegen][derivimplicit_solver]") {
 }
 
 
+SCENARIO("Some tests on euler solver", "[codegen][euler_solver]") {
+    GIVEN("A mod file with euler") {
+        std::string const nmodl_text = R"(
+            NEURON {
+                RANGE inf
+            }
+            INITIAL {
+                inf = 2
+            }
+            STATE {
+                n
+                m
+            }
+            BREAKPOINT {
+                SOLVE state METHOD euler
+            }
+            DERIVATIVE state {
+               m' = 2 * m
+               inf = inf * 3
+               n' = (2 + m - inf) * n
+            }
+        )";
+        THEN("Correct code is generated") {
+            auto const generated = get_cpp_code(nmodl_text);
+            std::string nrn_state_expected_code = R"(inst->Dm[id] = 2.0 * inst->m[id];
+            inf = inf * 3.0;
+            inst->Dn[id] = (2.0 + inst->m[id] - inf) * inst->n[id];
```

Comment:
Just to note here that with this change we start using the `Dvar` variables which we didn't before. This will introduce an extra unnecessary write and the `Dvar`s are part of the InstanceStruct and increase the memory usage of the mechanism.
We could improve this by avoiding the use of these variables, stop adding them to the InstanceStruct and just using local variables.
For the moment we can keep it like this for compatibility with NEURON.
cc: @pramodk

Reply:
Yes, perfectly fine with how you formulated!